### ISSUE TITLE
False positive in Rails/SaveBang

### ISSUE THREAD

**Issue by biow0lf** [bbatsov/rubocop#4593]
`Rails/SaveBang` cop claim on method `#save` in case conditions.

## Expected behavior

Rubocop should just ignore case when `#save` return value used in `case` switch.

## Actual behavior

`Use save! instead of save if the return value is not checked.`

## Steps to reproduce the problem

```ruby
class A
  def save
    true
  end
end
```

```ruby
class B
  def run
    case A.new.save
    when true
      # something
    when false
      # something else
  end
end
```

## RuboCop version

```
$ bundle exec rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by pocke** [comment]
Thanks for your reporting!


This cop has similar problems for the following code.

```ruby
a.save && foo
a.save || foo
```
